src/HtmlGenerator/Smart.Design.HtmlGenerator/Managers/ComponentDiscoverer.cs

```diff
@@ -0,0 +1,21 @@
+using System.Collections.Generic;
+using System.Linq;
+using Smart.Design.HtmlGenerator.Extensions;
+using Smart.Design.Razor.TagHelpers.Constants;
+
+namespace Smart.Design.HtmlGenerator.Managers
+{
+    public class ComponentDiscoverer : IComponentDiscoverer
+    {
+        /// <summary>
+        /// Returns a collection of the Smart design components implemented with Tag Helpers in Smart.Design.Razor library.
+        /// </summary>
+        /// <returns>A list of Tag Helpers rendering Smart design components</returns>
```

Comment:
In the file `GenerateOptions`, we talk about design elements but here we talk about Smart design components.
We need to decide on a term and use it consistently.